### PR TITLE
Improves fuzzers

### PR DESCRIPTION
**Pull request by Tpt**
Requires #109 to be fully functional but the code is independant from it.

### CODE REVIEW

**Review by arkpar — APPROVED** [review]

File: fuzz/fuzz_targets/refcounted_model.rs [hunk 1]
```diff
@@ -7,125 +7,77 @@
...
 				},
 		}
+	}
 
-		// We check the state
-		for (k, (_, v)) in &model {
-			assert_eq!(db.get(0, &[*k]).unwrap(), Some(vec![*v]));
+	fn map_operation(operation: &Operation) -> parity_db::Operation<Vec<u8>, Vec<u8>> {
+		match *operation {
+			Operation::Set(k) => parity_db::Operation::Set(vec![k], vec![k]),
```
Comment (cheme) [inline fuzz/fuzz_targets/refcounted_model.rs:69]: We could use different length of keys and values here, either with smallrng or directly with fuzzer input (with fuzzer input should only make sense for btree).
But that's probably already on your roadmap.

File: fuzz/fuzz_targets/simple_model.rs [hunk 2]
```diff
@@ -5,98 +5,59 @@
...
-
-			// We check the BTree backward iterator
-			model_content.reverse();
-			db_iter.seek_to_last().unwrap();
-			let mut db_content = Vec::new();
-			while let Some((k, v)) = db_iter.prev().unwrap() {
-				db_content.push((k[0], v[0]));
-			}
-			assert_eq!(db_content, model_content);
+	fn map_operation(operation: &(u8, Option<u8>)) -> parity_db::Operation<Vec<u8>, Vec<u8>> {
```
Comment (cheme) [inline fuzz/fuzz_targets/simple_model.rs:46]: Could have an util function for sharing this logic with ref counted (for now it's not very useful actually:)

**Review by cheme — APPROVED** [review]
:+1:

File: fuzz/fuzz_targets/refcounted_model.rs (hunk 1 above)
Comment (Tpt) [reply fuzz/fuzz_targets/refcounted_model.rs:69]: Yes, definitely. It's on my roadmap. The tricky thing is that we want keys to get often reused between operations.

Comment (cheme) [reply fuzz/fuzz_targets/refcounted_model.rs:69]: in my trie fuzzer I just for half the delete use a random index in the model

Comment (Tpt) [reply fuzz/fuzz_targets/refcounted_model.rs:69]: Sounds like a great idea! I was also thinking about generating from the fuzzer input a set of keys and then picking in this set when generating operations.

**Pull request merged**